src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/mapper/YFlowMapper.java

```diff
@@ -106,7 +106,7 @@ protected Collection<Flow> getDiverseWithFlow(Flow flow, FlowRepository flowRepo
         return flow.getDiverseGroupId() == null ? Collections.emptyList() :
                 flowRepository.findByDiverseGroupId(flow.getDiverseGroupId()).stream()
                         .filter(diverseFlow -> !flow.getFlowId().equals(diverseFlow.getFlowId())
-                                || (flow.getYFlowId() != null && !flow.getYFlowId().equals(diverseFlow.getYFlowId())))
+                                && (flow.getYFlowId() == null || !flow.getYFlowId().equals(diverseFlow.getYFlowId())))
```

Comment:
Maybe split into two filter calls to increase readability?

Reply:
Agree. Reformatted.